src/orange/surf/RecursiveSimplifier.hh

```diff
@@ -0,0 +1,98 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/surf/RecursiveSimplifier.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <utility>
+#include <variant>
+
+#include "SurfaceSimplifier.hh"
+#include "VariantSurface.hh"
+#include "detail/AllSurfaces.hh"
+#include "detail/RecursiveSimplifierImpl.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Recursively simplify and call the given function on the final surface.
```

Comment:
This would be a little more clear as:

Recursively simplify, then call the given function on the final surface.